app-connector/src/server/api.ts

```diff
@@ -0,0 +1,217 @@
+
+export class API {
+    private assureConnected() {
+        if (!connection.established()) {
+            return "Not connected to a kyma cluster, please re-connect"
+        }
+        return null
+    }
+    public findAll() {
+        return new Promise((resolve, reject) => {
+            let err = this.assureConnected()
+            if (err) {
+                reject(new Error(err));
+            }
+            request({
+                url: connection.info().metadataUrl,
+                method: "GET",
+                agentOptions: {
+                    cert: connection.certificate(),
+                    key: connection.privateKey()
+                },
+                rejectUnauthorized: !connection.info().insecure
+            }, (error: any, httpResponse: any, body: any) => {
+                if (error) {
+                    LOGGER.error("Error while Finding all APIs: %s", error)
+                    let resultobj = {
+                        statusCode: 500,
+                        body: error
+                    }
+                    reject(resultobj);
+                } else {
+                    let resultObj = { statusCode: httpResponse.statusCode, body: {} };
+                    if (httpResponse.statusCode >= 400) {
+                        let message = "Error while Finding all APIs: %s" + JSON.stringify(body, null, 2);
+                        LOGGER.error(message);
+                        resultObj.body = new Error(message)
+                        reject(resultObj);
+                    }
+                    else {
+                        LOGGER.debug("Received all APIs: %s", JSON.stringify(body, null, 2))
+                        resultObj.body = JSON.parse(body);
+                        resolve(resultObj);
+                    }
+                }
+            })
+        })
+
+
+    }
+
+    public create(serviceMetadata: any) {
+        return new Promise((resolve, reject) => {
+            let err = this.assureConnected()
+            if (err) {
+                reject(new Error(err));
+            }
+            request.post({
+                url: connection.info().metadataUrl,
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                json: serviceMetadata,
+                agentOptions: {
+                    cert: connection.certificate(),
+                    key: connection.privateKey()
+                },
+                rejectUnauthorized: !connection.info().insecure
+            }, (error: any, httpResponse: any, body: any) => {
+                if (error) {
+                    LOGGER.error("Error while Creating Api: %s", error)
+                    let resultobj = {
+                        statusCode: 500,
+                        body: error
+                    }
+                    reject(resultobj);
+                } else {
+                    let resultObj = { statusCode: httpResponse.statusCode, body: {} };

```

Comment:
Do not return status code, we are talking about a nice interface here, so either sucess or throw error so that client can react in a proper way.

Reply:
Reimplemented as we discussed